test-driver/src/Compare.ts

```diff
@@ -0,0 +1,215 @@
+ */
+
+import {OutputFormat} from './OutputFormat';
+import {Writer} from "ion-js";
+import {IonComparisonReport} from "./ComparisonReport";
+import {ComparisonContext} from "./ComparisonContext";
+import {IonCompareArgs} from "./CliCompareArgs";
+import {ComparisonResult, ComparisonResultType} from "ion-js";
+import {IonEvent, IonEventType} from "ion-js";
+import {IonTypes} from "ion-js";
+import {makeReader} from "ion-js";
+import {IonEventStream} from "ion-js";
+
+/**
+ * The `command`, `describe`, and `handler` exports below are part of the yargs command module API
+ * See: https://github.com/yargs/yargs/blob/master/docs/advanced.md#providing-a-command-module
+ */
+export const command = 'compare <input-file..>'
+
+export const describe = "Compare all inputs (which may contain Ion streams and/or EventStreams) against all other inputs " +
+    "using the Ion data model's definition of equality. Write a ComparisonReport to the output.";
+
+export const builder = {
+    'comparison-type': {
+        alias: "y",
+        default: 'basic',
+        choices: ['basic', 'equivs', 'non-equivs', 'equiv-timeline'],
+        describe: "Comparison semantics to be used with the compare command, from the set (basic | equivs | non-equivs |" +
+            "equiv-timeline). Any embedded streams in the inputs are compared for EventStream equality. 'basic' performs" +
+            "a standard data-model comparison between the corresponding events (or embedded streams) in the inputs." +
+            "'equivs' verifies that each value (or embedded stream) in a top-level sequence is equivalent to every other" +
+            "value (or embedded stream) in that sequence. 'non-equivs' does the same, but verifies that the values (or" +
+            "embedded streams) are not equivalent. 'equiv-timeline' is the same as 'equivs', except that when top-level" +
+            "sequences contain timestamp values, they are considered equivalent if they represent the same instant" +
+            "regardless of whether they are considered equivalent by the Ion data model. [default: basic]"
+    }
+}
+
+export const handler = function (argv) {
+    let args = new IonCompareArgs(argv);
+    new Compare(args);
+}
+
+/** Comparison semantics to be used with compare command */
+export enum ComparisonType {
+    BASIC = "basic",
+    EQUIVS = "equivs",
+    NON_EQUIVS ="non-equivs",
+    EQUIV_TIMELINE = "equiv-timeline"
+}
+
+/**
+ * Compare all inputs (which may contain Ion streams and/or EventStreams) against all other inputs
+ * using the Ion data model's definition of equality. Write a ComparisonReport to the output.
+**/
+export class Compare {
+
+    constructor(parsedArgs: IonCompareArgs) {
+        let output_writer = OutputFormat.createIonWriter(parsedArgs.getOutputFormatName());
+        if (output_writer) {
+            this.compareFiles(output_writer, parsedArgs);
+        }
+    }
+
+    compareFiles(ionOutputWriter: Writer, args: IonCompareArgs): void {
```

Comment:
Throughout the new `test-driver` code: please add some doc comments to the publicly visible methods. (Probably not necessary for simple `constructor`s or accessor methods.)

Reply:
Sure.